e2e/test_sanity.py

```diff
@@ -0,0 +1,120 @@
+"""
+Installs the vanilla distribution of kubeflow and validates the installation
+"""
+
+import os
+
+import pytest
+
+from kfp_server_api.exceptions import ApiException as KFPApiException
+from kubernetes.client.exceptions import ApiException as K8sApiException
+
+from e2e.constants import GENERIC_KUSTOMIZE_MANIFEST_PATH, DEFAULT_USER_NAMESPACE, CUSTOM_RESOURCE_TEMPLATES_FOLDER
+from e2e.utils import safe_open, wait_for, rand_name
+from e2e.config import metadata, region
+from e2e.cluster import cluster
```

Comment:
ah! I see how those fixtures are being called now. It was hard to see, usually you wouldn't expect a import to be doing this

thinking how to make this obvious and more customizable ...

Reply:
so the import calls the fixture and creates the cluster ? ya not readable

Reply:
the import imports the fixture

the cluster is created by pytest when the fixture is provided as an argument to the test function

if a fixture has other fixtures as dependencies, then those will also be created provided the fixture dependency is imported

For example

```
################# fixtures.py #################

@pytest.fixture
def cluster():
    return cluster

def kustomize(cluster):
    return kustomize

################# test_file.py ####################
from fixtures import cluster, kustomize

def test(kustomize):
    # pytest wants to create kustomize, but sees kustomize needs a cluster
    # pytest creates the cluster (using the cluster fixture that was imported), then creates kustomize

    # ... test logic
```